[Victoria281] iP

Victoria281 opened the pull request:
# Vic

> "Your mind is for having ideas, not holding them." – David Allen [(source)](https://gettingthingsdone.com/)

**Vic** frees your mind from remembering tasks and helps you stay organized. It's:

- **Text-based**
- **Easy to learn**
- ~~**FAST**~~ ⚡ **SUPER FAST** to use

All you need to do is:

1. [Download it from here](#).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you! 😉

And the best part? **It's FREE!**

## Features

- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon!)

If you are a Java programmer, you can use **Vic** to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Start using **Vic** today and experience the power of efficient task management! 🚀

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Victoria281 committed:
rename chatbot

Victoria281 committed:
Add echo feature

Victoria281 committed:
Add add, list feature

Victoria281 committed:
Add mark as done feature

Victoria281 committed:
Add ToDo, Event, Deadline class feature

Victoria281 committed:
Add automated text ui testing

Victoria281 committed:
Add handle errors feature

Victoria281 committed:
Update text ui testing to check exceptions

Victoria281 committed:
Add delete feature

Victoria281 committed:
Update text ui testing to check delete and task out of bounds

Victoria281 committed:
Convert commands into enums

pang-yy commented on src/main/java/Vic.java:
```diff
@@ -0,0 +1,237 @@
+                break;
+            case EVENT:
+                description = response.split(" ", 2)[1].split("/")[0];
+                if (description.length() < 1) throw new EmptyContentException();
```
I think it would be better to add braces `{}` for `if` statement.

Victoria281 replied:
Thanks for the reminder!

pang-yy commented on src/main/java/Vic.java:
```diff
@@ -0,0 +1,237 @@
+
+                String[] part2 = response.split("/");
+                String from = part2.length > 1 ? part2[1].split("from ").length > 1 ? part2[1].split("from ")[1] : "" : "";
+                if (from.length() < 1) throw new EmptyContentException();
```
I think it would be better to add braces `{}` for `if` statement.
I noticed the same issue in several other places too.

Victoria281 replied:
Thanks for the reminder!

pang-yy commented on src/main/java/Vic.java:
```diff
@@ -0,0 +1,237 @@
+        int taskID = -1;
+        try {
+            taskID = Integer.parseInt(option);
+        }
+        catch(NumberFormatException e) {
```
I think `catch(...) {` should be on the same line as `}`.
I noticed the same issue in several other places too.

Victoria281 replied:
I will take note. Thank you!

pang-yy commented on src/main/java/Vic.java:
```diff
@@ -0,0 +1,237 @@
+            return;
+        }
+        int taskID = parseTaskId(responseLst[1]);
+        if (taskID == -1) { return; }
```
I think `{` should have line break after it.

Victoria281 replied:
I missed that out! Thanks

pang-yy reviewed (commented):
Overall, I found your code easy to read for the most part, except a few places where the coding standard was not followed.
Good work! 👍

DanielJames0302 commented on src/main/java/Vic.java:
```diff
@@ -0,0 +1,237 @@
+            return;
+        }
+        int taskID = parseTaskId(responseLst[1]);
+        if (taskID == -1) { return; }
```
Perhaps follow Egyptian braces ?

Victoria281 replied:
Thanks for the reminder!

DanielJames0302 commented on src/main/java/Vic.java:
```diff
@@ -0,0 +1,237 @@
+                break;
+            case EVENT:
+                description = response.split(" ", 2)[1].split("/")[0];
+                if (description.length() < 1) throw new EmptyContentException();
```
Perhaps should follow Egyptain braces

Victoria281 replied:
Thanks for the reminder!

DanielJames0302 commented on src/main/java/Vic.java:
```diff
@@ -0,0 +1,237 @@
+            if (responseLst.length <= 1) throw new EmptyContentException();
+            Command command = Command.convertText(responseLst[0]);
+            switch (command) {
+            case TODO:
```
Well aware of the indentation of switch case

DanielJames0302 reviewed (commented):
LGTM. Just some nits to fix.

Victoria281 committed:
Check file exists and read from file

Victoria281 committed:
Delete and Edit actions saved in file

Victoria281 committed:
Handle corrupted file contents in storage file

Victoria281 committed:
Merge branch 'branch-Level-7'

Victoria281 committed:
Dates and times implemented

Victoria281 committed:
Merge branch 'branch-Level-8'

Victoria281 committed:
Structure project using OOP

Victoria281 committed:
Merge branch 'branch-A-MoreOOP'

Victoria281 committed:
Organize into Packages

Victoria281 committed:
Merge branch 'branch-A-Packages'

Victoria281 committed:
Merge branch 'branch-A-Gradle'

Victoria281 committed:
Reformatted code to follow javadoc and coding standards

Victoria281 committed:
Merge branch 'branch-A-JavaDoc'

Victoria281 committed:
Fixed some ui formatting

Victoria281 committed:
Set up and created JUnit tests for Parser class

Victoria281 committed:
Fixed and Wrote unit tests for storage function

Victoria281 committed:
Merge branch 'branch-A-JUnit'

Victoria281 committed:
Added dependencies to crate JAR file

Victoria281 committed:
Merge branch 'branch-A-Jar'

Victoria281 committed:
Fixed jar class name error

Victoria281 committed:
src/main/java

Victoria281 committed:
Find a task through keyword

Victoria281 committed:
Merge branch 'branch-Level-9'

Victoria281 committed:
Edit code styles to ensure coding standard

Victoria281 committed:
Merge branch 'branch-A-CheckStyle'

Victoria281 committed:
Convert vic chatbot into GUI with JavaFx

Victoria281 committed:
Design GUI styles

Victoria281 committed:
Merge branch 'branch-Level-10'

Victoria281 committed:
Update reading of input to use varargs

Victoria281 committed:
Merge branch 'branch-A-Varargs'

Victoria281 committed:
Add assertions in the Storage Class

The Storage class has areas in which proper initialisation of variables
is not checked, which may lead to potential issues.

Task data is correctly formatted during load and save operations
Task IDs is within range when splitting to read content

Assertions can catch errors early and will aid in performance of the
Storage class

Victoria281 committed:
Add assertions in the Parser Class

The Parser class is an important class that parses information, there
should be checks on the variables that are being sent to parse
information out of them

Add assertions to verify the integrity of the input data before parsing
Ensure that the parser also handles invalid inputs like out of range
data

This will help the parser catch invalid input and raise clearer error
messages

Victoria281 committed:
Update Javadocs comments and rename variables and functions

The following updates is made:
- edit class header comments to follow coding standards
- rename variables and functions for better clarity and understanding
  (toMarkDone becomes isMarkdown, toExit becomes isExit)

These changes help to make code more maintainable and clear with
adherence to project standards

Victoria281 committed:
Merge branch 'branch-A-Assertions'

Victoria281 committed:
Merge branch 'branch-A-CodeQuality'

Victoria281 committed:
Add extention to support tagging

Victoria281 committed:
Fix parsing of date for addition of tasks

Victoria281 committed:
Merge pull request #2 from Victoria281/branch-C-Tagging

Add extension from category C tagging

Victoria281 committed:
Include user guide and GUI of chatbot Vic

Victoria281 committed:
Merge branch 'branch-A-UserGuide'

Victoria281 committed:
Merge branch 'master' of github.com:Victoria281/ip
